src/accessibility.ts

```diff
@@ -10,6 +10,10 @@ export type Scenario = {
   /**
    * A function that returns the rendered output to test, or an Enzyme wrapper
    * created using Enzyme's `mount` function.
+   *
+   * Returning a VNode is preferred, as returning an Enzyme wrapper can have
+   * unexpected results if the top-level rendered component is a Fragment.
```

Comment:
Let's imagine this case:

```tsx
function MyComponent() {
  return (
    <>
      <label htmlFor="the-input">Label<span>
      <input type="text" id="the-input" />
    </>
  );
}

checkAccessibility({
  content: () => mount(<MyComponent />),
});
```

Internally, when an enzyme wrapper is provided, we do `wrapper.getDOMNode()` and mount that in an actual DOM container.

In this particular case, that will make the test fail, because only the `label` will be mounted, as `getDOMNode()` can only return one node, and the fragment is not a valid one.

On the other hand, if the component is passed as is, we internally create the enzyme wrapper attaching the result to the DOM (`mount(wrapper, { attachTo: container })`), which results in both the label and the input being properly mounted.

```tsx
checkAccessibility({
  content: () => <MyComponent />,
});
```

I didn't know how to synthesize this in a single sentence 😅 

Also, I can see other cases where this could also happen, like when a component returns an array of children instead of a fragment:

```tsx
function MyComponent() {
  return [
    <label key={1} htmlFor="the-input">Label<span>,
    <input key={2} type="text" id="the-input" />
  ];
}
```

Reply:
I have updated the comment to:

```
Returning a VNode is preferred. An Enzyme wrapper for a component with a
top-level Fragment will result in the test being run for the first DOM
child node only.
Use the Enzyme wrapper only if really needed, and consider wrapping your
component with a `<div />` in that case.
```